docs/guide/template-filters.md

```diff
@@ -0,0 +1,118 @@
+# Template Filters
+
+## Nunjucks supplied filters
+The Nunjucks templating language comes with several [built-in filters](https://mozilla.github.io/nunjucks/templating.html#filters). These filters apply functions to template data before outputting content to the page. They are called with a pipe operator `|` and can take arguments.
+```markup
+<h1>{{ data.page.headline | replace("foo", "bar" ) | upper }}</h1>
+```
+In this example, a hypothetical page headline will be retrieved from the page data and then piped to the 'replace' filter, along with two arguments. This filter will scan the incoming data and replace any instances of 'foo' with 'bar'. The output of this filter will then pass the result to the upper function, which will return the input as an uppercase string before being output to the page.
+
+>Note: The order of filters can be important, as they are applied sequentially to the input data.
+
+## Apostrophe supplied Filters
+There are several custom filters and filter sets (multiple, sequential filters) used in Apostrophe templating. Rather than have to remember each filter and the correct order, apostrophe exposes them automatically to the templates. In addition, Apostrophe also adds multiple "helper functions" which do not use the `| foo` filter syntax. These are documented with the modules that provide them. Several of the more common of these are within the documentation for the [util module](/reference/modules/util.html#template-helpers).
+
+## Custom template filters
+If you have special template needs you can also construct custom filters for use in Nunjucks templates.
+
+```javascript
+// lib/modules/link-widgets/index.js
+module.exports = {
+  extend: '@apostrophecms/widget-type',
+  options: {
+    label: 'Link to a Page'
+  },
+  fields: {
+    add: {
+      url: {
+        type: 'url',
+        label: 'URL',
+        required: true
+      },
+      label: {
+        type: 'string',
+        label: 'Label',
+        required: true
+      }
+    },
+    group: {
+      links: {
+        label: 'Links',
+        fields: [ 'url', 'label' ]
+      }
+    }
+  },
+  init(self) {
+    self.apos.template.addFilter({
+      stripHttp: self.stripHttp
+    });
+  },
+  methods(self) {
+    return {
+      stripHttp(s) {
+        return s.replace(/^(https?:|)\/\//, '');
+      }
+    };
+  }
+};
+```
+To use this filter you would simply pipe your data to the filter.
+
+```markup
+// lib/modules/link-widgets/views/widget.html
+<section data-link-widget>
+  <h2>{{ data.widget.label }}: {{ data.widget.url | stripHttp }}</h2>
```

Comment:
People who want to know how to make actual link widgets are going to find this, so you should include the `a` tag too.

Reply:
The purpose isn't self-explanatory. "By way of example, we'll create a link where the URL is included as part of the label, but keep just the interesting part."

Reply:
In addition to your suggestions, I also expanded on `addFilter` because it doesn't seem to be documented.